components-compose/src/main/java/uk/gov/hmrc/components/compose/organism/information/InformationMessageCardView.kt

```diff
@@ -0,0 +1,178 @@
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import uk.gov.hmrc.components.compose.R
+import uk.gov.hmrc.components.compose.atom.button.HmrcButton
+import uk.gov.hmrc.components.compose.organism.HmrcCardView
+import uk.gov.hmrc.components.compose.ui.extensions.enableTalkBackMergeAccessibility
+import uk.gov.hmrc.components.compose.ui.theme.HmrcAlwaysBlack
+import uk.gov.hmrc.components.compose.ui.theme.HmrcBlack
+import uk.gov.hmrc.components.compose.ui.theme.HmrcBlue
+import uk.gov.hmrc.components.compose.ui.theme.HmrcRed
+import uk.gov.hmrc.components.compose.ui.theme.HmrcRippleTheme
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme
+import uk.gov.hmrc.components.compose.ui.theme.HmrcWhite
+import uk.gov.hmrc.components.compose.ui.theme.HmrcYellow
+
+enum class MessageType(val backgroundColor: Color, val headlineTint: Color, val iconColor: Color, val icon: Int) {
+    WARNING(HmrcYellow, HmrcAlwaysBlack, HmrcAlwaysBlack, R.drawable.components_ic_warning),
+    INFO(HmrcBlue, HmrcWhite, HmrcWhite, R.drawable.ic_info),
+    URGENT(HmrcRed, HmrcWhite, HmrcWhite, R.drawable.ic_info),
+    NOTICE(HmrcBlack, HmrcWhite, HmrcWhite, R.drawable.components_ic_warning)
+}
+
+@Composable
+fun InformationMessageCardView(
+    modifier: Modifier = Modifier,
+    headline: String,
+    headlineContentDescription: String = "",
+    text: String? = null,
+    messageType: MessageType,
+    buttons: List<InformationMessageButton> = emptyList()
+) {
+    val textColor: Color by remember { mutableStateOf(messageType.headlineTint) }
+    val iconColor: Color by remember { mutableStateOf(messageType.iconColor) }
+    val backgroundColor: Color by remember { mutableStateOf(messageType.backgroundColor) }
+    val icon: Int by remember { mutableIntStateOf(messageType.icon) }
+
+    HmrcCardView(modifier = modifier, customBackgroundColor = backgroundColor) {
+        Row(
+            modifier = modifier
+                .fillMaxWidth()
+                .enableTalkBackMergeAccessibility()
+                .padding(HmrcTheme.dimensions.hmrcSpacing16),
+        ) {
+            Image(
+                modifier = Modifier.size(HmrcTheme.dimensions.hmrcIconSize36),
+                painter = painterResource(id = icon),
+                contentDescription = "",
+                colorFilter = ColorFilter.tint(iconColor)
+            )
+            Spacer(modifier = Modifier.width(HmrcTheme.dimensions.hmrcSpacing8))
+            Text(
+                text = headline,
+                modifier = Modifier
+                    .weight(1f)
+                    .heightIn(HmrcTheme.dimensions.hmrcIconSize36)
```

Comment:
Do we need this? Would this restrict the size of the content if we had a lot of text?

Reply:
This is taken from the WarningView and it doesn't display the icon/text correctly without it.